back/spec/support/stub_env_helper.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module StubEnvHelper
+  # @overload stub_env(var_name, value)
+  #  @param var_name [String] the name of the environment variable
+  #  @param value [String] the value of the environment variable
+  # @overload stub_env(hash)
+  #  @param hash [Hash] a hash of environment variables and their values
+  #
+  # @example Stub a single environment variable
+  #   stub_env('FOO', 'bar')
+  # @example Stub multiple environment variables
+  #   stub_env({'FOO' => 'bar', 'BAZ' => 'qux'})
+  def stub_env(*args)
+    allow(ENV).to receive(:[]).and_call_original
+    allow(ENV).to receive(:fetch).and_call_original
+
+    if args.size == 1 && args.first.respond_to?(:each_pair)
+      args.first.each_pair do |var_name, value|
+        stub_single_env(var_name, value)
+      end
+    elsif args.size == 2
+      stub_single_env(args.first, args.second)
+    else
+      raise ArgumentError, 'Invalid arguments for stub_env'
+    end
+  end
```

Comment:
Maybe always having the argument as keyword attributes would make the implementation a bit simpler and the API is as clear as before. Not critical of course.

```suggestion
  # @example Stub environment variables
  #   stub_env(FOO: 'bar')
  def stub_env(env_vars)
    allow(ENV).to receive(:[]).and_call_original
    allow(ENV).to receive(:fetch).and_call_original

    env_vars.each_pair do |var_name, value|
      stub_single_env(var_name, value)
    end
  end
```

Reply:
In this case, I think it's mostly a matter of personal preference and I took some extra care to write clear documentation with examples.